metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/utils/ElasticsearchUtils.java

```diff
@@ -111,19 +111,24 @@ public static TransportClient getClient(Map<String, Object> globalConfiguration,
     Settings.Builder settingsBuilder = Settings.builder();
     settingsBuilder.put("cluster.name", globalConfiguration.get("es.clustername"));
     settingsBuilder.put("client.transport.ping_timeout","500s");
+    settingsBuilder.put("transport.type", "security4");
+    Object xPackUser = globalConfiguration.get("es.xpackuser");
+    if (xPackUser != null) {
+      settingsBuilder.put("xpack.security.user", xPackUser);
+    }
     if (optionalSettings != null) {
       settingsBuilder.put(optionalSettings);
     }
     Settings settings = settingsBuilder.build();
-    TransportClient client;
+    PreBuiltXPackTransportClient client;
```

Comment:
First question I have is will this work with and without XPack enabled?

Reply:
yep. it might make sense to also check for a empty string for the "es.xpackuser" config, or does globalConfiguration.get return null also for empty strings?